src/controller.rs

```diff
@@ -114,27 +120,69 @@ async fn reconcile(app_instance: Arc<AppInstance>, ctx: Arc<Context>) -> Result<
         serde_json::from_value(config.metadata.get(PACK_KEY).unwrap().clone())
             .map_err(Error::DecodeKubecfgPackageMetadata)?;
 
-    launch_job(&app_instance, &kubecfg_pack_metadata, Arc::clone(&ctx)).await?;
+    let configmap_name = create_configmap_overlay(&app_instance, Arc::clone(&ctx)).await?;
+    launch_job(
+        &app_instance,
+        &kubecfg_pack_metadata,
+        configmap_name,
+        Arc::clone(&ctx),
+    )
+    .await?;
```

Comment:
I was stumped on how to do this without writing to a `ConfigMap` as a go-between with the `overlay.json` and spawned `Job`, as without the `ConfigMap` (external state) the pods created from the `Job` have no understanding of what overlay to apply as it was within the controller loop itself. So I settled on doing this here for now.

Happy to alter it though :rocket: